src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+                # For nested dicts, initialize any dicts that don't already exist.
+                if element not in data:
+                    data[element] = {}
+            data = data[element]
+        return full_object
+
+    @staticmethod
+    def _serialize_rescue(rescue: Rescue,
+                          attr_filter: typing.Optional[typing.Set[str]] = None) -> typing.Dict:
+        """
+        Serialize a Rescue object into a Dict that follows the JSONAPI format.
+
+        Args:
+            rescue (Rescue): The Rescue object to serialize.
+
+        Returns:
+            A Dict containing the data from the Rescue, in JSONAPI format.
+        """
+        # Associate each Rescue attribute with a 'path' in the JSON object.
+        # The value can either be a string with the JSON path, or a tuple
+        # with the JSON path and the value to serialize.
+        # If given only the JSON path, it will automatically grab the value
+        # from the rescue object.
+        maps = {
+            'api_id': ('id', f"{rescue.api_id}"),
+            'client': 'attributes.client',
+            'code_red': 'attributes.codeRed',
+            'platform': ('attributes.platform', rescue.platform.name.casefold()),
+            'status': ('attributes.status', rescue.status.name.casefold()),
+            'system': 'attributes.system',
+            'title': 'attributes.title',
+            'outcome': 'attributes.outcome',
+            'unidentified_rats': 'attributes.unidentifiedRats',
+            'lang_id': ('attributes.data.langID', rescue.lang_id.casefold()),
+            'irc_nickname': 'attributes.data.IRCNick',
+            'board_index': 'attributes.data.boardIndex',
+            # Not sure what this attribute is, but we'll include it anyway.
+            'data_status': ('attributes.data.status', {}),
+            'quotes': ('attributes.quotes', [
+                {'author': quote.author,
+                 'message': quote.message,
+                 'createdAt': quote.created_at.astimezone().isoformat(),
+                 'updatedAt': quote.updated_at.astimezone().isoformat(),
+                 'lastAuthor': quote.last_author}
+                for quote in rescue.quotes
+            ]),
+            'marked_for_deletion': ('attributes.data.markedForDeletion',
+                                    {'marked': rescue.marked_for_deletion.marked,
+                                     'reason': rescue.marked_for_deletion.reason,
+                                     'reporter': rescue.marked_for_deletion.reporter
+                                     }
+                                    ),
+            'rats': ('relationships.rats.data',
+                     [{'type': 'rats', 'id': f"{rat.uuid}"} for rat in rescue.rats]),
+            'first_limpet': ('relationships.firstLimpet.data',
+                             {'type': 'rats',
+                              'id': f"{rescue.first_limpet}"} if rescue.first_limpet else None)
+        }
```

Comment:
I believe the objective of the map  /partial object serialization was to avoid initializing an entire ApiRescue data structure whenever we wanted to make an update. Hence i suggested using factory methods, so we could differ evaluation til the point we actually needed to initialize a partial data structure.

Reply:
Potential solution in c1dd9b4 for review.